### ISSUE TITLE
Support downloading the manifest in other compressed formats

### ISSUE THREAD

**Issue by gsnedders** [web-platform-tests/wpt#16081]
Per discussion on IRC:

11:16 < jgraham> https://github.com/web-platform-tests/wpt-actions-test/releases/tag/merge_commit_01d8001cd1172d07239760d8be2cf980ff743924 I made some compressed manifests for you, if you want to benchmark download and extract time (all made using the latest version of the relevant tool on Ubuntu using the highest compression setting but without additional tricks like prebuilt dictionaries)
11:50 < gsnedders> jgraham: I did this locally yseterday ;P
11:50 < gsnedders> (prebuilt dictionaries are unlikely to be a big gain with the size of data we have)
11:51 < jgraham> gsnedders: The point was "on GitHub" to allow a more accurate picture of download + decompress time
11:51 < gsnedders> sure, but I expect a lot of that depends on proximity to edge nodes wherever GitHub host the thing and connection speed
11:52 < gsnedders> certainly I'm concerned enough about the slow connection case that I'd e.g. say bz2 is worth it
11:52 < gsnedders> but I also work-from-train relatively frequently
11:52 < jgraham> Of course, but it at least allows you to test reasonable cases
11:53 < jgraham> And you can probably set up throttling if you want to check the work from (wherever) case
11:53 < gsnedders> (Also question of what hardware we're decompressing on; nothing I have is that recent nowadays given how slow perf gains have been with CPUs…)
11:56 < jgraham> gsnedders: At least on my system bzip2 takes 300ms to decompress compared to 90ms for gzip and maybe 35 for zstd (not sure I got that right)
11:57 < jgraham> And 50ms for brotli
11:58 < gsnedders> 280ms v. 70ms v. 30ms here
11:58 < jgraham> So yeah, I'm pretty skeptical about taking on 200-250ms of overhead on the decompress time
11:59 < gsnedders> So I think we need to always have at least gz or bz2 given we can rely on Python being able to decode them
11:59 < gsnedders> IIRC, bz2 is worth it if the download is < 16Mbps
12:02 < gsnedders> and I only just get over that on my home connection (VDSL2, sync'd at 76.6Mbps), presumably msotly due to the cost of TCP ramp-up
12:03 < gsnedders> downloading it multiple times, I often don't download speeds over 16Mbps, which implies we *do* want the cost of decompressing it
12:04 < jgraham> Yeah, so on my computer, bz2 seems to be a small win compared to gz, but anything else would be a large win (total for gz is 752ms, bz2 is 559, zst is 315ms
12:04 < gsnedders> yeah, zstd is a clear win, but we need an extension module or calling out to it as a subprocess for it
12:04 < jgraham> Based on one trial, writing to dev/null, etc. So not science
12:04 < jgraham> But indicative
12:04 < gsnedders> like I'm happy to say we should use zstd if we can
12:05 < gsnedders> and I think I favour bz2 > gzip
12:05 < jgraham> Yeah, I think it would be worthwhile to try installing that and fall back to bz2 if it's not available. For compression we can simply make it available.
12:07 < gsnedders> I'm skeptical of the value of dictionaries, given it'd be another thing we need to fetch, and we're a large file
12:57 < gsnedders> (I really want to throw together some evolutionary algorithm to find some maxima for compression ratio for zstd with all the advance options!)

To summarise: we want to support bzip2 and zstd.

- [x] Support building bzip2
- [x] Support building zstd
- [ ] Support downloading bzip2
- [ ] Support downloading zstd

We probably want some period where we're building all three (gzip, bzip2, and zstd), though then we can go down to only building bzip2 and zstd.

**Comment by foolip** [comment]
This was fixed in https://github.com/web-platform-tests/wpt/pull/16136.

**Comment by foolip** [comment]
> We probably want some period where we're building all three (gzip, bzip2, and zstd), though then we can go down to only building bzip2 and zstd.

I would like to keep gzip indefinitely, because it simplifies any tooling that needs all manifests if you can always assume that exists.
